### DISCUSSION TITLE
Can I use this library on another microcontroller?

### DISCUSSION THREAD

**Discussion by FrancescoCrevatin** [jpb10/SolarCalculator#4] [General]
Hi jpb10,
first of all: great job.

Meanwhile, I want to tell you that I used the library for an Arduino project and everything worked fine, only I found some problems with the "automatic" time calculation using: setTime (toUtc (compileTime ())) ;, so I used a clock (RTC) that gives time to the other function you created: setTime (hr, min, sec, day, mo, yr).

But I wanted to ask you, since I'm not very grasped in C programming and using libraries: Can I use the same library to write a program on another microcontroller? That is, I would need the same library in C, and it seems to me that it is in C, but I wanted to ask you for confirmation if I could use the exact same library on a microcontroller other than the Arduino family.

Thanks again for your availability.

**Comment by jpb10** [comment]
Here's the C version: **[CSolarCalculator](https://github.com/jpb10/CSolarCalculator/)**

Usage is slightly different:

```
JulianDay jd = jdFromUnix(utc);
calcSunriseSunset(jd, latitude, longitude, &transit, &sunrise, &sunset);
```

Notice the `&` before each output argument.

**Reply by Macintosh-Fan** [reply]
Just wondering, are there certain features that are not available in the C version which are in the C++ version (except for the `&` thing)?

**Reply by jpb10** [reply]
It's not currently up-to-date, but otherwise it's the exact same program.

**Comment by FrancescoCrevatin** [comment]
Thank you @jpb10. I'll try next days.
Thanks again for all the work you have done
